src/services/GasStation.ts

```diff
@@ -0,0 +1,148 @@
+import { Context } from "fastmcp";
+import { config } from "../config.js";
+
+export class GasStation {
+  protected readonly headers: Record<string, string>;
+  private readonly gasStationEndpoint: string;
+  private readonly context: Context<any>;
+
+  constructor(context: Context<any>, network: "testnet" | "mainnet") {
+    if (!config.aptos_build.botKey) {
+      throw new Error(
+        `APTOS_BOT_KEY is not set. To generate a Bot Key: 
+        1. Go to [https://build.aptoslabs.com/](https://build.aptoslabs.com/)
+        2. Click on your name in the bottom left corner
+        3. Click on "Bot Keys"
+        4. Click on the "Create Bot Key" button
+        5. Copy the Bot Key and paste it into the MCP configuration file as an env arg: APTOS_BOT_KEY=<your-bot-key>`
+      );
+    }
+    this.gasStationEndpoint =
+      network === "testnet"
+        ? config.gas_station.testnetUrl
+        : config.gas_station.mainnetUrl;
+    this.headers = {
+      Authorization: `Bearer ${config.aptos_build.botKey}`,
+      "x-is-aptos-bot": "true",
+      "Content-Type": "application/json",
+    };
+    this.context = context;
+  }
+
+  async createGasStation({
+    organization_id,
+    project_id,
+    application_id,
+  }: {
+    organization_id: string;
+    project_id: string;
+    application_id: string;
+  }) {
+    try {
+      const appHeaders = {
+        "x-jwt-organization-id": organization_id,
+        "x-jwt-project-id": project_id,
+        "x-jwt-application-id": application_id,
+      };
+
+      const gasStationResponse = await fetch(
+        `${this.gasStationEndpoint}/admin/application`,
```

Comment:
Would it be easier to use the admin client? https://github.com/aptos-labs/gas-station/tree/main/packages/gas-station-admin-client. We could release it publicly.

Reply:
probably, but for now it is only 2 calls so can live with it =)